pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -883,7 +883,7 @@ private void updatePublishDispatcher(Optional<Policies> optPolicies) {
 
         //topic-level policy is not set, try to use namespace-level rate policy
         final String clusterName = brokerService.pulsar().getConfiguration().getClusterName();
-        final PublishRate publishRate = policies.publishMaxMessageRate != null
+        final PublishRate publishRate = policies !=null && policies.publishMaxMessageRate != null
```

Comment:
Why would this policies be null?  Do you have any case?

Reply:
There is no NPE case in the production environment,But I see that the logic of the code below is judged to be null.
[// attach the resource-group level rate limiters, if set
        String rgName = policies != null && policies.resource_group_name != null
          ? policies.resource_group_name
          : null;]